### PR TITLE
Node consistency 41 50

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose

The following PR addresses nodes' names and descriptions to make them more descriptive for beginners.

**41) NODE: Color Green**
-CHANGES: 
Output: integer
 Output Description: Green value for RGB color model, int between 0 and 255 inclusive.

**42) NODE: Color Blue**
-CHANGES: 
Output: integer
Output Description: Blue value for RGB color model, int between 0 and 255 inclusive.

**43) NODE: Color.Alpha**
-CHANGES: 
Output: integer
Output Description: Hue value for color as double between 0 and 1 inclusive.

**44) NODE: Color.Hue**
-CHANGES: 
Output: double
Output Description: Hue value for color as double between 0 and 1 inclusive.

**45) NODE: Color.Components**
-CHANGES: 
Input: color
Outputs: alpha red green blue

**46) NODE: String.TrimWhitespace**
-CHANGES: 
Change input and output from str to string

**47) NODE: String.TrimTrailingWhitespace**
-CHANGES: 
Change input and output from str to string

**48) NODE: String.TrimLeadingWhitespace**
-CHANGES: 
Change input and output from str to string

**49) NODE: String.ToUpper**
-CHANGES: 
Change input and output from str to string

**50) NODE: String.ToLower**
-CHANGES: 
Change input and output from str to string

image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers
@SHKnudsen
@Amoursol 
@QilongTang 

### FYIs
-No FYI

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@martinstacey Can you run test `DSCoreNodesTests.ColorTests.TestComponents` locally to make sure it actually pass? Saw that is the regression related to this PR but not sure if it is a real one

File: src/Libraries/CoreNodes/Color.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public byte Red
         /// <summary>
         ///     Find the green component of a color, 0 to 255.
         /// </summary>
-        /// <returns name="green">int between 0 and 255 inclusive.</returns>
+        /// <returns name="integer">Green value for RGB color model, int between 0 and 255 inclusive.</returns>
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodes/Color.cs:32]: `int`

Comment (aparajit-pratap) [reply src/Libraries/CoreNodes/Color.cs:32]: you can say "integer" in the description.

Comment (QilongTang) [reply src/Libraries/CoreNodes/Color.cs:32]: For this one I saw we sometimes use `int` sometimes use `integer` in code base, do we want to unify that?

Comment (Amoursol) [reply src/Libraries/CoreNodes/Color.cs:32]: We also use the term `number` and `double` interchangeably. I'm in favour of the more verbose calls here towards `integer` over int for overall clarity.

Comment (aparajit-pratap) [reply src/Libraries/CoreNodes/Color.cs:32]: I prefer int, double, bool, and string like it's in most languages but that's just me.

Comment (Amoursol) [reply src/Libraries/CoreNodes/Color.cs:32]: I have set up a poll here (https://forum.dynamobim.com/t/request-for-feedback-language-preferences/55066) to gauge the interest of the Dynamo Community in this.

Comment (Amoursol) [reply src/Libraries/CoreNodes/Color.cs:32]: @martinstacey - Poll ran to a conclusion. Please follow pattern explained inside of Slack channel. Copied here for historical record: 

After internal communication and a public poll on the Dynamo Forums: https://forum.dynamobim.com/t/request-for-feedback-language-preferences/55066, can we please run with the following amendments to all submitted nodes:

- Please use short-form terms for all input port names for only commonly used DesignScript base type abbreviations, and long-form terms in the hover description. Short form terms taken from  https://dynamobim.org/wp-content/links/DesignScriptGuide.pdf
- This means things such as **func** or **comb** `do` become long-form. But things such as **bool** or **int** stay short-form.

Comment (martinstacey) [reply src/Libraries/CoreNodes/Color.cs]: addressed in 8252853

**Comment by martinstacey** [comment]
> @martinstacey Can you run test `DSCoreNodesTests.ColorTests.TestComponents` locally to make sure it actually pass? Saw that is the regression related to this PR but not sure if it is a real one

@QilongTang 
The test was not passing because the names for outputs in node `Color.Components` were changed. I just updated the test to use the new output names in 7b5b605. 

image

**Review by Amoursol — APPROVED** [review]

**Pull request merged**